From 1cfb60b53b87f8f46973087a25913b00b126b61f Mon Sep 17 00:00:00 2001
From: Kurt Kiefer <kurt.kiefer@arthrex.com>
Date: Mon, 17 Jan 2022 13:01:15 -0800
Subject: [PATCH 1/1] mark new devices as ready for capture

This fixes an issue where gstreamer first tries to probe the input
capabilities and emits numerous warning when it is unable to do so.
---
 v4l2loopback.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/v4l2loopback.c b/v4l2loopback.c
index 8e7cfcd..9958a6e 100644
--- a/v4l2loopback.c
+++ b/v4l2loopback.c
@@ -1107,6 +1107,7 @@ static int vidioc_s_fmt_out(struct file *file, void *priv,
 		dev->buffer_size = PAGE_ALIGN(dev->pix_format.sizeimage);
 		fmt->fmt.pix.sizeimage = dev->buffer_size;
 		allocate_buffers(dev);
+		dev->ready_for_capture = 1;
 	}
 	return ret;
 }
-- 
2.17.1

